extensions/notebook/src/extension.ts

```diff
@@ -25,6 +26,11 @@ let controller: JupyterController;
 type ChooseCellType = { label: string, id: CellType };
 
 export async function activate(extensionContext: vscode.ExtensionContext): Promise<IExtensionApi> {
+
+	const bookTreeViewProvider = new BookTreeViewProvider(vscode.workspace.rootPath || '');
```

Comment:
>vscode.workspace.rootPath [](start = 55, length = 25)

Just to check - vscode.workspace isn't undefined even if you don't have a workspace opened?

Reply:
It's always there - it's a core API. rootPath might be empty / null if there's no open workspace.